src/background/set-timeout-manager.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AlarmUtils } from 'background/alarm-utils';
+import { WindowUtils } from 'common/window-utils';
+
+export class SetTimeoutManager {
+    constructor(private utils: { alarmUtils?: AlarmUtils; windowUtils?: WindowUtils }) {}
+
+    public setTimeout(callback: Function, delayInMs: number, name?: string): void {
+        if (name && this.utils.alarmUtils) {
+            this.utils.alarmUtils.createAlarmWithCallback(name, delayInMs, callback);
+        } else {
+            this.utils.windowUtils?.setTimeout(callback, delayInMs);
+        }
```

Comment:
nit: Should we check that windowUtils is not null, just in case? Maybe add the check and log an error when neither the alarm or window utils are non-null?